packages/api/src/domain/__tests__/user-token.ts

```diff
@@ -1,14 +1,27 @@
+import { nanoid } from "nanoid";
 import { UserToken } from "../user-token";
+import { v4 as uuidv4 } from "uuid";
 
-export function createUserToken(token: Partial<UserToken>): UserToken {
-  return UserToken.build({
-    token: token.token || "",
-    cxId: token.cxId || "",
-    userId: token.userId || "",
-    expiryTime: token.expiryTime || 0,
-    oauthRequestToken: token.oauthRequestToken || "",
-    oauthRequestSecret: token.oauthRequestSecret || "",
-    oauthUserAccessToken: token.oauthUserAccessToken || "",
-    oauthUserAccessSecret: token.oauthUserAccessSecret || "",
-  });
+export const makeUserToken = (token: Partial<UserToken>): UserToken => {
+  return {
+    token: token.token ?? nanoid(),
+    cxId: token.cxId ?? uuidv4(),
+    userId: token.userId ?? uuidv4(),
+    expiryTime: token.expiryTime ?? getPastTimestamp(),
+    oauthRequestToken: token.oauthRequestToken ?? "",
+    oauthRequestSecret: token.oauthRequestSecret ?? "",
+    oauthUserAccessToken: token.oauthUserAccessToken ?? nanoid(),
+    oauthUserAccessSecret: token.oauthUserAccessSecret ?? nanoid(),
+    clone: function (): UserToken {
+      return makeUserToken(this);
+    },
+  };
+};
+
+function getPastTimestamp(): number {
+  const currentDate = new Date();
+  const pastDate = new Date(currentDate);
+  pastDate.setDate(currentDate.getDate() - 1);
+
+  return Math.floor(pastDate.getTime() / 1000);
 }
```

Comment:
Maybe we could replace this by `faker.date.past().getTime()`?

If its something more specific, like returning a past timestamp in seconds, then a dedicated, _shared_ function would be nice, with the name indicating the unit/precision (seconds).

But if we really want to return the last millisecond, then a function here makes sense. I'd just remove the first date creation, or just simplify altogether with `new Date().getTime() - 1`.